packages/dataviews/src/components/dataviews/stories/index.story.tsx

```diff
@@ -183,39 +186,30 @@ function PlanetOverview( { planets }: { planets: SpaceObject[] } ) {
  * or pagination controls.
  */
 export const FreeComposition = () => {
-	const view = {
-		type: 'table' as const,
-		search: '',
-		page: 1,
-		perPage: 10,
-		layout: {},
-		filters: [],
-	};
-
-	const paginationInfo = {
-		totalItems: data.length,
-		totalPages: 1,
-	};
-
-	const { data: shownData } = useMemo( () => {
-		return filterSortAndPaginate( data, DEFAULT_VIEW, fields );
-	}, [] );
-
-	const planets = shownData.filter( ( item ) =>
+	const [ view, setView ] = useState< View >( {
+		...DEFAULT_VIEW,
+		fields: [ 'categories' ],
+		titleField: 'title',
+		descriptionField: 'description',
+		mediaField: 'image',
+	} );
+
+	const { data: processedData, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( data, view, fields );
+	}, [ view ] );
+
+	const planets = processedData.filter( ( item ) =>
 		item.categories.includes( 'Planet' )
 	);
 
-	// id` is a number in the data, but a string in the DataViews component.
-	const getItemId = ( item: SpaceObject ) => item.id.toString();
-
 	return (
 		<DataViews
-			data={ shownData }
-			getItemId={ getItemId }
-			view={ view }
-			onChangeView={ () => {} }
-			fields={ fields }
+			getItemId={ ( item ) => item.id.toString() }
 			paginationInfo={ paginationInfo }
+			data={ processedData }
+			view={ view }
+			fields={ [] }
```

Comment:
Your view includes a "categories", "title", "description", and "image" fields, but you don't pass the definition of these fields. It doesn't impact anything today because you're not rendering the layout/view yet. I just wanted to note it :)